eng/README.md

```diff
@@ -0,0 +1,50 @@
+# `eng` directory
+
+The `eng` directory contains source code for automated tooling running on this repository pull requests.
+
+For context on this directory, see [Design guidelines for spec repos validation tooling] (Microsoft-internal).
+
+## Code conventions
+
+Below are code convention we strive to follow in `eng` directory:
+
+### package.json
+
+- We align `package.json` dependencies versions across all `package.json` files.
+- We align `package.json` dependencies numbers with [microsoft/typespec package.json].
+  In few cases we allow more frequent update cadence.
+- We avoid doing package overrides. For example, we use `v8` of `eslint` instead of `v9` to avoid an override.
+  [See this comment for details][eslint override].
+- We order `package.json` keys as follows: `name private type main bin scripts engines dependencies devDependencies`.
+
+### package-lock.json
+
+- We maintain only top-level `package-lock.json` file. Running `npm install` from top-level dir removes the need
+  to ever have other files.
+- We ensure the lock file remains clean by ensuring that a PR that adds or modifies any `package.json` dependencies,
+  makes changes equivalent to following protocol:
+  - `cd <local-specs-clone-root>`
+  - `git clean -xdf` to remove all untracked files.
+  - Copy-over [`package-lock.json` from `main`] to local clone.
+  - `npm install` to reflect the added or modified dependencies.
+- In case any dependencies have been removed from any `package.json`, we do `rm package-lock.json` and `npm install`.  
+  This way we ensure the lock file remains free of unused dependencies.
+- We do `npm update` only in stand-alone PRs.
+- To avoid inconsistent content of `package-lock.json` due to bugs like [npm/cli #7384],
+  we ensure to use the same node and npm version to update `package-lock.json`.  
+  As of 7/15/2024 this is `node 20.15.1` and `npm 10.7.0`.
```

Comment:
```suggestion
- To avoid inconsistent content of `package-lock.json` due to bugs like [npm/cli #7384],
  we ensure to use the same node and npm version to update `package-lock.json`.  
  As of 7/15/2024 this is `node 20.15.1` and `npm 10.7.0`.
```